edge-hub/src/Microsoft.Azure.Devices.Routing.Core/RouteResult.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Routing.Core
+{
+    using System;
+
+    public class RouteResult : IEquatable<RouteResult>
+    {
+        public RouteResult(Endpoint endpoint, uint priority, uint timeToLiveSecs)
+        {
+            this.Endpoint = endpoint;
+            this.Priority = priority;
+            this.TimeToLiveSecs = timeToLiveSecs;
+        }
+
+        public readonly Endpoint Endpoint;
+        public readonly uint Priority;
+        public readonly uint TimeToLiveSecs;
+
+        public bool Equals(RouteResult other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            // Check endpoint, priority, and TTL
+            bool areEqual =
+                this.Endpoint.Equals(other.Endpoint) &&
+                this.Priority == other.Priority &&
+                this.TimeToLiveSecs == other.TimeToLiveSecs;
+
+            return ReferenceEquals(this, other) || areEqual;

```

Comment:
When would references be equal but not the inner values?

Reply:
Suggest to move ReferenceEquals check to its own if block and before line 27.

Reply:
yes, the ref check should be before property check